### PR TITLE
Changelog entry for v0.10.0

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

This PR adds the changelog entry for v0.10.0.

**Preview** at https://pygmt-dev--2651.org.readthedocs.build/en/2651/changes.html

Related to issue #2640

Previous release v0.9.0 at #2469

**TODO**

- [x] Copy the draft release note from https://github.com/GenericMappingTools/pygmt/releases to `doc/changes.md`
- [x] Update the DOI badge
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt and `CITATION.cff`
- [x] Add the version to the compatibility table https://github.com/GenericMappingTools/pygmt#compatibility-with-gmtpythonnumpy-versions
- [ ] Add highlights
- [x] Update with any remaining PRs merged before release
- [ ] Edit the change list to remove any trivial changes (updates to the README, typo fixes, CI configuration, etc)
- [x] Edit the list of people who contributed to the release, linking to their GitHub account. Sort their names by the number of commits made since the last release (e.g., use `git shortlog HEAD...v0.1.2 -sne`).

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: CITATION.cff
Comment (seisman) [inline CITATION.cff:83]: Version information at line 84 is not updated.

File: doc/changes.md
Comment (yvonnefroehlich) [inline doc/changes.md:24]:
```suggestion
* Figure.colorbar: Add alias for "Q" ([#2608](https://github.com/GenericMappingTools/pygmt/pull/2608))
* Figure.image: Add alias for "G" ([#2615](https://github.com/GenericMappingTools/pygmt/pull/2615))
* Figure.meca:  Add aliases for "L", "T", and "Fr" ([#2546](https://github.com/GenericMappingTools/pygmt/pull/2546))
* pygmt.grdfill: Add alias for "N" ([#2618](https://github.com/GenericMappingTools/pygmt/pull/2618))
* pygmt.select: Add aliases for "C", "F", and "L" ([#2466(https://github.com/GenericMappingTools/pygmt/pull/2466)))
* Allow passing RGB xarray.DataArray images into grdimage ([#2590](https://github.com/GenericMappingTools/pygmt/pull/2590))
* Better handling of optional virtual files (e.g., shading in Figure.grdimage) ([#2493](https://github.com/GenericMappingTools/pygmt/pull/2493))
* clib.Session: Wrap the GMT_Get_Common API function ([#2500](https://github.com/GenericMappingTools/pygmt/pull/2500))
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:11]: Suggestions for highlights?

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:58]: Should this go under 'Enhancements' instead of 'Maintenance'?

File: README.rst
Comment (weiji14) [inline README.rst:184]: Do we keep the same order list for the authors? Output of commits made on main branch at 1d8c0fdfd1c986147641aaf0cdb4e9e47af988ae to first commit at e3c542a173ad6f74d5072326d07275662249f288 (`git shortlog 1d8c0f...e3c542a -sne`) shows this after I merge the same author names:

```
544  Dongdong Tian
255  Leonardo Uieda
216  Wei Ji
191  Will Schlitzer
169  dependabot[bot]
130  Yvonne Fröhlich
124  Michael Grund
109  Max Jones
 33  Yao Jiayuan
 15  Liam Toney
  7  yohaimagen
  4  Tong
  3  kmaterna
  2  Abhishek Anant
  2  Amanda Leaman
  2  Andre L. Belem
  2  Andrey Shmakov
  2  Claudio Satriano
  2  Conor Bacon
  2  Jamie J Quinn
  2  MichaeINeumann
  2  Noor Buchi
  2  Shivani chauhan
  2  Tyler Newton
  2  alperen-kilic
  2  carocamargo
  2  daroari
  2  obaney
  2  srijac
  1  Alicia Ngoc Diep Ha
  1  Andrés Ignacio Torres
  1  Becky Salvage
  1  Brook
  1  Cody Woodson
  1  Emily McMullan
  1  Jack Beagley
  1  Julius Busecke 
  1  Kadatatlu Kishore
  1  Loïc Houpert
  1  Malte Ziebarth
  1  Mark Wieczorek
  1  Philipp Loose
  1  Sean H
  1  Soham Banerjee
  1  Tawanda
  1  Vitor Gratiere Torres
  1  mdtanker
  1  shahid-0
  1  teriyakiSauce23
```

File: doc/changes.md
Comment (seisman) [reply doc/changes.md:11]: Merge the two entries in the "New Features" category and move it to "Highlights"?

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:25]:
```suggestion
* Figure.grdimage: Allow passing RGB xarray.DataArray images ([#2590](https://github.com/GenericMappingTools/pygmt/pull/2590))
```

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:31]:
```suggestion
* Figure.text: Remove the deprecated 'incols' parameter (deprecated since v0.8.0) ([#2473](https://github.com/GenericMappingTools/pygmt/pull/2473))
```

File: doc/changes.md
Comment (seisman) [reply doc/changes.md:58]: Yes, I agree.

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:66]: Move it down:
```suggestion
```

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:74]:
```suggestion
* CI: Use mamba-org/provision-with-micromamba to setup micromamba ([#2435](https://github.com/GenericMappingTools/pygmt/pull/2435))
* CI: Migrate provision-with-micromamba to setup-micromamba ([#2536](https://github.com/GenericMappingTools/pygmt/pull/2536))
```

File: README.rst
Comment (seisman) [reply README.rst:184]: Based on the counts of commits, I think we should at least make two changes:

1. Move "Yvonne Fröhlich" before "Michael Grund"
2. Move "Yao Jiayuan" before "Liam Toney"

As for the order between Leo and me, I'm not sure about this, considering that Leo created this package.

File: doc/changes.md
Comment (weiji14) [reply doc/changes.md]: Ok, have combined the two non-ASCII character PRs into one highlight at 81dfe9dbddcf90cb4a5e06537b3f888780abc485. Any other ones, or this is good enough?

File: README.rst
Comment (weiji14) [reply README.rst:184]: > Move "Yvonne Fröhlich" before "Michael Grund"

Ok with this. I've been meaning to ask @yvonnefroehlich if she would like to be a maintainer actually, she's been helping a lot with the documentation and doing a lot of good quality reviews lately.

> Move "Yao Jiayuan" before "Liam Toney"

I don't think either of them have made a commit since 2021 (< v0.4.1), so not sure why we should change the order now... Liam did do quite a bit of work on the ROSES 2021 and 2020 videos, and I it was somewhat agreed at https://github.com/GenericMappingTools/pygmt/pull/1414#discussion_r683333747 to keep the order this way?

> As for the order between Leo and me, I'm not sure about this, considering that Leo created this package.

Leo said at https://github.com/GenericMappingTools/pygmt/pull/1594#discussion_r740248554 that he was happy for you to go first, and I'd say you deserve it more now :slightly_smiling_face: But we can also leave it if you feel shy, and put your name first on the paper :wink:

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:42]: TODO add #2383 once merged.

```suggestion
* Add tutorial to explain naming of PyGMT figure elements ([#2383](https://github.com/GenericMappingTools/pygmt/pull/2383))
* Add tutorial to show interactive data visualization via 'panel' ([#2498](https://github.com/GenericMappingTools/pygmt/pull/2498))
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:11]: Update count also.
```suggestion
* Three new tutorials and seven new gallery examples
```

File: README.rst
Comment (seisman) [reply README.rst:184]: > > Move "Yvonne Fröhlich" before "Michael Grund"
> 
> Ok with this. I've been meaning to ask @yvonnefroehlich if she would like to be a maintainer actually, she's been helping a lot with the documentation and doing a lot of good quality reviews lately.

Yes, she's well qualified. 

> > Move "Yao Jiayuan" before "Liam Toney"
> 
> I don't think either of them have made a commit since 2021 (< v0.4.1), so not sure why we should change the order now... Liam did do quite a bit of work on the ROSES 2021 and 2020 videos, and I it was somewhat agreed at [#1414 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1414#discussion_r683333747) to keep the order this way?

Sounds reasonable. Let's keep it unchanged.
 
> > As for the order between Leo and me, I'm not sure about this, considering that Leo created this package.
> 
> Leo said at [#1594 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1594#discussion_r740248554) that he was happy for you to go first, and I'd say you deserve it more now 🙂 But we can also leave it if you feel shy, and put your name first on the paper 😉

Aha, I didn't realize that. It's my honor to be listed as the first one. BTW, you're really good at digging history 😉

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:35]: Minor change so that user won't take "geopandas" as a PyGMT package/module/function:
```suggestion
* Geopandas integration: Mapping int/int64 to int32 for OGR_GMT format ([#2592](https://github.com/GenericMappingTools/pygmt/pull/2592))
```

File: doc/changes.md
Comment (seisman) [inline doc/changes.md:37]: Users may be confused since GMT 6.5.0 is not released yet.
```suggestion
* Figure.meca: Fix beachball offsetting for ndarray input (requires GMT>6.4.0) ([#2576](https://github.com/GenericMappingTools/pygmt/pull/2576))
```

File: README.rst
Comment (weiji14) [reply README.rst]: Ok, done at 852b025b26e14d6b5e327858a4312591560ef914

File: doc/changes.md
Comment (michaelgrund) [inline doc/changes.md:43]:
```suggestion
* Add tutorial to show interactive data visualization via ``panel`` ([#2498](https://github.com/GenericMappingTools/pygmt/pull/2498))
```

File: CITATION.cff
Comment (weiji14) [inline CITATION.cff:13]: Btw, we should remind @leouieda to change his affilliation - https://www.leouieda.com/blog/back-to-brazil.html. Let's do it after the release.

File: doc/changes.md
Comment (michaelgrund) [inline doc/changes.md:29]:
```suggestion
* Figure.text: Remove the deprecated "incols" parameter (deprecated since v0.8.0) ([#2473](https://github.com/GenericMappingTools/pygmt/pull/2473))
```
To be consistent with the other points.

**Review by michaelgrund — APPROVED** [review]
Looks good!

File: doc/changes.md
Comment (yvonnefroehlich) [inline doc/changes.md:37]:
```suggestion
* Figure.meca: the "scale" parameter can accept int/float/str values ([#2566](https://github.com/GenericMappingTools/pygmt/pull/2566))
```

File: doc/changes.md
Comment (yvonnefroehlich) [inline doc/changes.md:18]:
```suggestion
* Figure.meca: Add aliases for "L", "T", and "Fr" ([#2546](https://github.com/GenericMappingTools/pygmt/pull/2546))
```

File: doc/changes.md
Comment (yvonnefroehlich) [inline doc/changes.md:48]:
```suggestion
* Add gallery example showing how to build an envelope around a curve ([#2587](https://github.com/GenericMappingTools/pygmt/pull/2587))
```

File: doc/changes.md
Comment (yvonnefroehlich) [inline doc/changes.md:44]: It seems like single quotation marks are enough here (see output of  the lines 50 and 51):
```suggestion
* Add tutorial to show interactive data visualization via `panel` ([#2498](https://github.com/GenericMappingTools/pygmt/pull/2498))
```
single_back_quotation_marks

File: README.rst
Comment (yvonnefroehlich) [reply README.rst]: >>> Move "Yvonne Fröhlich" before "Michael Grund"

>> Ok with this. I've been meaning to ask @yvonnefroehlich if she would like to be a maintainer actually, she's been helping a lot with the documentation and doing a lot of good quality reviews lately.

> Yes, she's well qualified.

First of all thanks a lot for this positive feedback, @weiji14 and @seisman :blush:. I'm glad to hear this! I am really thankful for contributing to this project; I learned a lot. So, I would be pleased to become a maintainer of PyGMT. Hopefully the other @GenericMappingTools/pygmt-maintainers are also fine with this.
I feel I should have the time to help with the documentation and reviewing PRs in the future, but for more complicated code-related tasks I would need more (free) time, which is currently not very likely because of the workload of my PhD projects.

Just one comment regarding the commit list above: Looking at the commit numbers of the single contributors at https://github.com/GenericMappingTools/pygmt/graphs/contributors there is partly a significant difference to `git shortlog 1d8c0f...e3c542a -sne` (I can reproduce @weiji14 numbers). Looking at the total number of commits made so far to this repro, it seems that GitHub additionally includes / counts co-authorship?

File: doc/changes.md
Comment (yvonnefroehlich) [reply doc/changes.md]: Hm, a third highlight would be nice, but so far I don't have one 🙁. So, I am OK with two highlights,

File: CITATION.cff
Comment (weiji14) [inline CITATION.cff:79]:
```suggestion
date-released: 2023-09-02
```

File: doc/changes.md
Comment (weiji14) [inline doc/changes.md:3]:
```suggestion
## Release v0.10.0 (2023/09/02)
```

**Review by weiji14 — COMMENTED** [review]
Pushing out the release date to 2nd Sep (following UTC time).

**Comment by seisman** [comment]
Looks good to me, but do we want to move "Yvonne Fröhlich" as mentioned in https://github.com/GenericMappingTools/pygmt/pull/2651#discussion_r1312431397

**Comment by weiji14** [comment]
> Looks good to me, but do we want to move "Yvonne Fröhlich" as mentioned in [#2651 (comment)](https://github.com/GenericMappingTools/pygmt/pull/2651#discussion_r1312431397)

Oh yes, let me do that.

File: README.rst
Comment (weiji14) [reply README.rst]: > > > > Move "Yvonne Fröhlich" before "Michael Grund"
> 
> > > Ok with this. I've been meaning to ask @yvonnefroehlich if she would like to be a maintainer actually, she's been helping a lot with the documentation and doing a lot of good quality reviews lately.
> 
> > Yes, she's well qualified.
> 
> First of all thanks a lot for this positive feedback, @weiji14 and @seisman 😊. I'm glad to hear this! I am really thankful for contributing to this project; I learned a lot. So, I would be pleased to become a maintainer of PyGMT. Hopefully the other @GenericMappingTools/pygmt-maintainers are also fine with this. I feel I should have the time to help with the documentation and reviewing PRs in the future, but for more complicated code-related tasks I would need more (free) time, which is currently not very likely because of the workload of my PhD projects.

No worries, feel free to help out as much (or as little) as you can, PhD comes first of course! If you can open up a PR to modify [team.md](https://github.com/GenericMappingTools/pygmt/blob/main/doc/team.md), we can continue on from there.

> Just one comment regarding the commit list above: Looking at the commit numbers of the single contributors at https://github.com/GenericMappingTools/pygmt/graphs/contributors there is partly a significant difference to `git shortlog 1d8c0f...e3c542a -sne` (I can reproduce @weiji14 numbers). Looking at the total number of commits made so far to this repro, it seems that GitHub additionally includes / counts co-authorship?

Yes, it might be that GitHub includes co-authored commits, whereas `git log -sne` only counts the main author.

**Review by seisman — APPROVED** [review]

**Comment by weiji14** [comment]
Thanks everyone for the reviews! Let me merge this and tag the release before my next flight in one hour :laughing:

**Comment by michaelgrund** [comment]
> Thanks everyone for the reviews! Let me merge this and tag the release before my next flight in one hour 😆

Great work! Thanks all for your efforts!

**Comment by yvonnefroehlich** [comment]
> > Thanks everyone for the reviews! Let me merge this and tag the release before my next flight in one hour 😆
> 
> Great work! Thanks all for your efforts!

Thanks a lot @weiji14 for handling this release :slightly_smiling_face: and all other for their contributions to make this release possible :tada:!

File: README.rst
Comment (yvonnefroehlich) [reply README.rst]: > No worries, feel free to help out as much (or as little) as you can, PhD comes first of course!

Sounds good 🙂!

> If you can open up a PR to modify [team.md](https://github.com/GenericMappingTools/pygmt/blob/main/doc/team.md), we can continue on from there.

Please see PR #2655.

**Pull request merged**
